### ISSUE TITLE
libEGL.so.1: undefined symbol

### ISSUE THREAD

**Issue by henningphan** [knapsu/plex-media-player-appimage#11]
Hello,
Thank you for awesome work.

I got a bug when running 2.6 and 2.7.

**This is what happens when I try to run the appimage:**
`/tmp/.mount_Plex_M5jHpLC/usr/bin//plexmediaplayer.wrapper: line 125:  2015 Aborted                 LD_LIBRARY_PATH="" zenity --question --title="$TITLE" --text="$TEXT" 2> /dev/null
/tmp/.mount_Plex_M5jHpLC/usr/bin//plexmediaplayer.wrapper: line 125:  2019 Aborted                 LD_LIBRARY_PATH="" zenity --question --title="$TITLE" --text="$TEXT" 2> /dev/null
/tmp/.mount_Plex_M5jHpLC/usr/bin/plexmediaplayer: symbol lookup error: /tmp/.mount_Plex_M5jHpLC/usr/lib/libEGL.so.1: undefined symbol: drmGetDevice2`

**Comment by probonopd** [comment]
Also see https://github.com/anupam-git/vlc-appimage/issues/5 -- does it work if you delete `libEGL.so.1` from the AppImage?

**Comment by henningphan** [comment]
How do I delete it? No prior experience with appimage.

**Comment by probonopd** [comment]
```
# Extract the contents of the AppImage
./VLC*.AppImage --appimage-extract
# Delete libEGL
find ./squashfs-root -type f -name "libEGL.so.1" -delete
# See if the runs
./squashfs-root/AppRun
```

(typed from memory, slight changes might be needed to the `find` command but you should get the idea)

**Comment by henningphan** [comment]
@probonopd 
Confirmed that your solution works.

How does one put the extract back? Or does that require special appimage tools?

**Comment by knapsu** [comment]
@probonopd should I exclude `libEGL.so` when assembling the AppImage? What do you think?

**Comment by probonopd** [comment]
Yes, looks like it. cc @TheAssassin

**Comment by TheAssassin** [comment]
It's a graphics library acting as some kind of interface between OpenGL and X11, IIRC. It should be available on most systems, so I'd say it's safe to exclude it. I didn't verify this, though.

@probonopd we should exclude it upstream.

**Comment by probonopd** [comment]
According to https://github.com/AppImage/AppImages/search?utf8=%E2%9C%93&q=libEGL.so.1&type= there are systems which need the bundled libEGL.so.1. Maybe we need to use a similar workaround like for libstdc++ (check if a newer version comes with the system; use the bundled one otherwise)?

**Comment by TheAssassin** [comment]
@probonopd I wonder which library pulls in libEGL, it's normally not a direct dependency. Maybe the lib that depends on EGL can be excluded?

**Comment by probonopd** [comment]
Good point!

There seem to be some library combinations/clusters that should be treated as "one", we need to find out which ones those are.

**Comment by henningphan** [comment]
Any update on this? The latest version was still not working

**Comment by cchalk** [comment]
Hi,

I downloaded the latest version 2.11.0.867-f27f8d2a
But libEGL.so.1 is still included in the AppImage.

I did the following, then it worked great:

`./Plex_Media_Player_2.11.0.867-f27f8d2a_x64.AppImage --appimage-extract`
`cd squashfs-root`
`rm usr/lib/libEGL.so.1`
`../appimagetool-x86_64.AppImage .`

Then a new AppImage is created and runs fine. appimagetool is from:

[https://github.com/AppImage/AppImageKit/releases](url)

**Comment by knapsu** [comment]
Thanks @cchalk for the info. As `libEGL` is expected to be on all systems there is no point in keeping it in the AppImage when it generates compatibility issues. Will blacklist it.

**Comment by knapsu** [comment]
Fixed in 464e29bba508eb1c04f1be5fd0ad71d66e29d1b7
